src/components/article/PostArticle.jsx

```diff
@@ -0,0 +1,147 @@
+    }, (error, result) => {
+      if (result.event === "success") {
+        this.setState({ imageUrl: result.info.secure_url });
+
+      }
+    });
+  }
+
+  formInputHelper = (field, inputPlaceHolder, inputName, inputType, onChange, inputValue, inputClass, inputRows) => {
+    let data;
+    if (field === "textInput") {
+      data = (
+        <input
+          placeholder={inputPlaceHolder}
+          name={inputName}
+          type={inputType}
+          onChange={onChange}
+          value={inputValue}
+          className={inputClass}
+          required
+        />
+      );
+    } else if (field === "textArea") {
+      data = (
+        <textarea
+          placeholder={inputPlaceHolder}
+          rows={inputRows}
+          name={inputName}
+          onChange={onChange}
+          value={inputValue}
+          className={inputClass}
+          required
+        />
+      );
+    }
+    return data;
+  }
+
+  formOnClickHelper = (inputPlaceHolder, inputName, inputType, onClick, inputValue, inputClass) => (
+    <input
+      placeholder={inputPlaceHolder}
+      name={inputName}
+      type={inputType}
+      onClick={onClick}
+      value={inputValue}
+      className={inputClass}
+      readOnly
+    />
+  );
+
+  render() {
+    const { title, imageUrl, description, body, audioUrl, tags } = this.state;
+    return (
+      <main className="mt-5 pt-5">
+        <div className="container">
+          <section className="mt-4">
+            <form onSubmit={this.onSubmit}>
+              <div className="text-center">
+                <div className="form-group shadow-textarea">
+                  {this.formInputHelper("textInput", "Article Title", "title", "text", this.onChange, title, "form-control font-weight-bold no-border in-font", null)}
+                  {this.formInputHelper("textInput", "Article Description", "description", "text", this.onChange, description, "form-control font-weight-bold no-border", null)}
+                  {this.formInputHelper("textArea", "Article body...", "body", this.onChange, null, body, "form-control z-depth-1 no-border in-fontt", "20")}
+                  {this.formOnClickHelper("Choose Article Image Url(Optional)", "imageUrl", "text", this.handleOnClick, imageUrl, "form-control no-border")}
+                  {this.formInputHelper("textInput", "Article Audio Url(Optional)", "audioUrl", "text", this.onChange, audioUrl, "form-control no-border", null)}
+                </div>
+              </div>
+              <TagsInput value={tags} onChange={this.onChang} />
+              <button type="submit" id="post-article" className="btn btn-primary">
+                Publish
+              </button>
+            </form>
+          </section>
+        </div>
+      </main>
+    );
+  }
+}
+
+PostArticle.propTypes = {
+  createArticles: propTypes.func.isRequired
+};
+
+export default connect(
+  null,
+  {
+    createArticles
+  }
+)(PostArticle);
```

Comment:
Could you add a new line at the end of the file

Reply:
@shakirandagire I have worked on this